### PR TITLE
Easier to use sr_movements: using the joint name and HandCommander

### PR DESCRIPTION
**Pull request by ugocupcic**
Instead of remapping the topics / having to specify correct min and max for each joint, we now have the possibility to simply specify the joint name, the HandCommander does the rest (fetch min / max, topics etc...). Kept the "old version" in place as well for compatibility reason.

### CODE REVIEW

File: sr_movements/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,11 @@ include_directories( ${ImageMagick_INCLUDE_DIRS})
 
 rosbuild_add_executable(sr_movements src/sr_movements.cpp src/movement_from_image.cpp src/partial_movement.cpp src/movement_publisher.cpp)
 target_link_libraries(sr_movements ${ImageMagick_LIBRARIES})
+target_link_libraries(sr_movements HandCommander)
```
Comment (toliver) [inline sr_movements/CMakeLists.txt]: This line should be removed, as explained in (http://www.ros.org/wiki/rosbuild/CMakeLists/Examples#Linking_against_libraries_built_by_other_ROS_packages).
HandCommander library is already being exported in the sr_hand package manifest.

File: sr_movements/CMakeLists.txt [hunk 2]
```diff
@@ -21,6 +21,11 @@ include_directories( ${ImageMagick_INCLUDE_DIRS})
...
+
+rosbuild_add_rostest(test/test_movements.test)
+rosbuild_add_rostest(test/test_joint_name_movement.test)
 
 rosbuild_add_library(sr_movement src/movement_from_image.cpp src/partial_movement.cpp src/movement_publisher.cpp)
-target_link_libraries(sr_movement ${ImageMagick_LIBRARIES})
\ No newline at end of file
+target_link_libraries(sr_movement ${ImageMagick_LIBRARIES})
+target_link_libraries(sr_movement HandCommander)
```
Comment (toliver) [inline sr_movements/CMakeLists.txt]: Same as above

File: sr_movements/src/movement_publisher.cpp [hunk 3]
```diff
@@ -167,6 +216,11 @@
   {
     pub = publisher;
   }
+
+  void MovementPublisher::set_subscriber(ros::Subscriber Subscriber)
```
Comment (toliver) [inline sr_movements/src/movement_publisher.cpp]: Initial lower case is our usual convention for function parameter names.

Comment (ugocupcic) [inline sr_movements/src/movement_publisher.cpp]: fixed.

File: sr_movements/test/test_joint_name_movement.test [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    <param name="nb_step" value="1000"/>
+    <param name="msg_type" value="sr"/>
+    <param name="testing" value="true"/>
+  </node>
+
+  <param name="hztest1/topic" value="/sh_ffj3_mixed_position_velocity_controller/command" />
+  <param name="hztest1/hz" value="100.0" />
+  <param name="hztest1/hzerror" value="0.5" />
+  <param name="hztest1/test_duration" value="5.0" />
+  <test test-name="hztest_test" pkg="rostest" type="hztest" name="hztest1" />
```
Comment (toliver) [inline sr_movements/test/test_joint_name_movement.test]: I assume that test tags are not allowed to contain param tags as node do?

Comment (ugocupcic) [inline sr_movements/test/test_joint_name_movement.test]: Haven't tested it, it's the way it's written on the ROS wiki so I didn't look further. Let me try it.

Comment (ugocupcic) [inline sr_movements/test/test_joint_name_movement.test]: OK pushed new version.

**Pull request merged**